src/NewsService.Business/Commands/News/GetNewsCommand.cs

```diff
@@ -0,0 +1,141 @@
+    private readonly INewsResponseMapper _mapper;
+    private readonly IRequestClient<IGetUsersDataRequest> _rcGetUsers;
+    private readonly IRequestClient<IGetImagesRequest> _rcGetImages;
+    private readonly IUserInfoMapper _userInfoMapper;
+    private readonly IResponseCreator _responseCreator;
+    private readonly ITagsInfoMapper _tagsInfoMapper;
+    private readonly IChannelRepository _channelRepository;
+    private readonly ITagsRepository _tagsRepository;
+    private readonly IChannelInfoMapper _channelInfoMapper;
+    private readonly ILogger<GetNewsCommand> _logger;
+
+    private async Task<List<UserData>> GetUsersDataAsync(List<Guid> usersIds, List<string> errors)
+    {
+      if (usersIds is null || !usersIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IGetUsersDataRequest, IGetUsersDataResponse>(
+          _rcGetUsers,
+          IGetUsersDataRequest.CreateObj(usersIds),
+          errors,
+          _logger))
+        .UsersData;
+    }
+
+    private async Task<List<ImageData>> GetUsersAvatarsAsync(List<Guid> imagesIds, List<string> errors)
+    {
+      if (imagesIds is null || !imagesIds.Any())
+      {
+        return null;
+      }
+
+      return (await RequestHandler.ProcessRequest<IGetImagesRequest, IGetImagesResponse>(
+          _rcGetImages,
+          IGetImagesRequest.CreateObj(imagesIds, ImageSource.User),
+          errors,
+          _logger))
+        .ImagesData;
+    }
+
+    public GetNewsCommand(
+      INewsRepository newsRepository,
+      INewsResponseMapper mapper,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      IUserInfoMapper userInfoMapper,
+      IResponseCreator responseCreator,
+      ITagsInfoMapper tagsInfoMapper,
+      IChannelRepository channelRepository,
+      ITagsRepository tagsRepository,
+      IChannelInfoMapper channelInfoMapper,
+      ILogger<GetNewsCommand> logger)
+    {
+      _newsRepository = newsRepository;
+      _mapper = mapper;
+      _rcGetUsers = rcGetUsers;
+      _rcGetImages = rcGetImages;
+      _userInfoMapper = userInfoMapper;
+      _responseCreator = responseCreator;
+      _tagsInfoMapper = tagsInfoMapper;
+      _channelRepository = channelRepository;
+      _tagsRepository = tagsRepository;
+      _channelInfoMapper = channelInfoMapper;
+      _logger = logger;
+    }
+
+    public async Task<OperationResultResponse<NewsResponse>> ExecuteAsync(Guid newsId)
+    {
+      OperationResultResponse<NewsResponse> response = new();
+
+      DbNews dbNews = await _newsRepository.GetAsync(newsId);
+      if (dbNews is null)
+      {
+        return _responseCreator.CreateFailureResponse<NewsResponse>(
+          HttpStatusCode.NotFound);
+      }
+
+      List<UserData> usersData =
+        await GetUsersDataAsync(
+          new List<Guid>() 
+          { dbNews.PublishedBy.GetValueOrDefault(), dbNews.CreatedBy },
```

Comment:
you add wrong value from PublishedBy. Add only if PublishedBy is not null

Reply:
fixed